tests/commoninterop.nim

```diff
@@ -260,7 +260,7 @@ proc commonInteropTests*(name: string, swCreator: SwitchCreator) =
           check line == test
           testFuture.complete(line)
         except CancelledError as e:
-          raise e
+          raise newException(CancelledError, "Native handler cancelled: " & e.msg, e)
```

Comment:
Cancel errors should probably be re-raised without modification to maintain the call stack.